.github/workflows/qa.yaml

```diff
@@ -94,13 +94,13 @@ jobs:
         run: |
           set -eu
 
-          go test -coverpkg=./... -coverprofile=/tmp/coverage.out -covermode=set ./... -shuffle=on
+          ./gotestcov ./...
```

Comment:
Reusing the script is nice, but we also inherit then some side effect, like python webserver hosting. If we want to reuse it in CI, we should have a strip-down mode just doing what’s needed from the script.

Reply:
and btw, we are loosing `-shuffle=on` with that change.

Reply:
I agree that the Python server hosting does not really belong here. I kept it just so our team could update their versions with the new one. How do you feel about creating a `gotestcov-ci` with all the "local" functionalities stripped out? That way we can still keep the updated script as an artifact.

How much do we care about `-shuffle=on`? I saw that Go has it disabled by default. It mentions that it saves the seed but, unless we save it on the CI, it's actually worse for us to have it enabled I think (or does the toolchain report the seed when running the tests?).

Reply:
> I agree that the Python server hosting does not really belong here. I kept it just so our team could update their versions with the new one. How do you feel about creating a gotestcov-ci with all the "local" functionalities stripped out? That way we can still keep the updated script as an artifact.

Shouldn’t that be in a reusable action then, in the desktop-engineering repo.

> How much do we care about -shuffle=on? I saw that Go has it disabled by default. It mentions that it saves the seed but, unless we save it on the CI, it's actually worse for us to have it enabled I think (or does the toolchain report the seed when running the tests?).

From the documentation: `If -shuffle is set to on, then it will seed the randomizer using the system clock`. So, there is nothing to save, we want the tests to run in random order. Saving the seed could be used at some point, but as we already run them all in parallel, it’s heavily depending on the cpu load.

Reply:
> Shouldn’t that be in a reusable action then, in the desktop-engineering repo.

The script has Rust related stuff as well, so I'm not really sure it should be added there (at least not in its current state). I think we can iterate more about making it more "dynamic" and then creating an action there, but I don't think it's worth blocking the current PR on that (as it sounds more like a "Friday afternoon" type of work).

> So, there is nothing to save, we want the tests to run in random order.

Yeah, but if the tests fail due to running in a specific order, we would like to be able to reproduce the ordering, no? Also, I kind of like the way it runs the tests now by running the package tests before running the integration tests (even though I know this is probably due to alphabetic ordering :laughing: ), it makes a lot of sense.

> as we already run them all in parallel, it’s heavily depending on the cpu load.

I'm not sure if I understand exactly how the parallelization of the tests works, but I thought it worked on a per-package basis (i.e. run in parallel all the tests (that can be executed this way) in the package and only then proceed to run the tests in the next package and so on).